pennylane/pulse/transmon.py

```diff
@@ -0,0 +1,193 @@
+
+    where :math:`[a^\dagger_p, a_q] = i \delta_{pq}` are bosonic creation and annihilation operators.
+    The first term describes the dressed qubit frequencies :math:`\omega_q`, the second term their coupling :math:`g_{ij}` and the last the anharmonicity :math:`\delta_q`,
+    which all can vary for different qubits. In practice, the bosonic operators are restricted to a finite dimension of the local Hilbert space (default ``d=2`` corresponds to qubits).
+    In that case, the anharmonicity is set to :math:`delta=0` and ignored.
+
+    .. note:: Currently only supporting ``d=2`` with qudit support planned in the future.
+
+    TODO: resource / source for good values.
+
+    .. seealso::
+
+        :func:`~.drive`
+
+    Args:
+        connections (list[tuple(int)]): List of connections ``(i, j)`` between qubits i and j.
+        omega (Union[float, list[float]]): List of dressed qubit frequencies in GHz.
+            When passing a single float all qubits are assumed to have that same frequency.
+        g (Union[float, list[float]]): List of coupling strengths in GHz. Needs to match the length of ``connections``.
+            When passing a single float all connections are assumed to have the same strength.
+        delta (Union[float, list[float]]): List of anharmonicities in GHz. Ignored when ``d=2``.
+            When passing a single float all qubits are assumed to have that same anharmonicity.
+        wires (list): Optional, defaults to the unique wires in ``connections`` when set to ``None``. When passing explicit ``wires``,
+            needs to at least contain all unique wires in ``connections`` and can be used to initiate additional, unconnected qubits.
+        d (int): Local Hilbert space dimension. Defaults to ``d=2`` and is currently the only supported value.
+
+    Returns:
+        HardwareHamiltonian: a :class:`~.ParametrizedHamiltonian` representing the transmon interaction
+
+    **Example**
+
+    We can set up the transmon interaction Hamiltonian with uniform coefficients by passing ``float`` values.
+
+    .. code-block::
+
+        connections = [[0, 1], [1, 3], [2, 1], [4, 5]]
+        H = qml.pulse.transmon_interaction(connections, omega=0.5, g=1.)
+
+    The resulting :class:`~.ParametrizedHamiltonian` consists of ``4`` coupling terms and ``6`` qubit terms.
+
+    >>> print(H)
+    ParametrizedHamiltonian: terms=10
+
+    """
+    if d != 2:
+        raise NotImplementedError(
+            "Currently only supporting qubits. Qutrits and qudits are planned in the future."
+        )
+
+    if wires is not None and not all(i in wires for i in qml.math.unique(connections)):
+        raise ValueError(
+            f"There are wires in connections {connections} that are not in the provided wires {wires}"
+        )
+
+    wires = wires or qml.math.unique(connections)
+    n_wires = len(wires)
+
+    # Prepare coefficients
+    # TODO: make coefficients callable / trainable. Currently not supported
```

Comment:
It might actually work, it's not supported atm in the sense that it is not tested yet

Reply:
The problem currently is that HardwareHamiltonian by default uses the reordering. Maybe something to change in a next PR since this one is a bit bloated